api_endpoints.py

```diff
@@ -92,8 +92,11 @@ def post(self):
         if s == "**":
             s = key
 
+        x_real_ip = self.request.headers.get("X-Real-IP")
```

Comment:
This case should already be handled by Tornado's HTTPServer. Is xheaders disabled in your prod branch?